src/site/markdown/server/manage-guice-distributed-james.md

```diff
@@ -260,3 +261,102 @@ by setting the parameter `elasticsearch.index.mailbox.name` to the name of your
+
+## Setting Cassandra user permissions
+
+With a Cassandra server that serving multiple applications including James. 
+The keyspace of each James server need to be protected from another, 
+that also means, by the credential provided for a James server, 
+I cannot use it to access or modify other keyspaces. 
+
+We recommend you to not use the default Cassandra credential and provide 
+a different one with a minimized permissions for each application. 
+
+### Prerequisites
+
+We're gonna use the super user to create roles and grant permissions for them. 
+To do that, Cassandra requires you to login via username/password authentication 
+and enable granting in cassandra configuration file.
+
+For example:
+```
+echo -e "\nauthenticator: PasswordAuthenticator" >> /etc/cassandra/cassandra.yaml
+echo -e "\nauthorizer: org.apache.cassandra.auth.CassandraAuthorizer" >> /etc/cassandra/cassandra.yaml
+```
+
+### Prepare Cassandra roles & keyspaces for James  
+
+#### Create a role
+
+Visit [cassandra documentation](http://cassandra.apache.org/doc/latest/cql/security.html) section `CREATE ROLE` for more information
+
+E.g.
+```
+CREATE ROLE linagora WITH PASSWORD = 'linagora' AND LOGIN = true;
+```
+#### Create a keyspace
+
+Visit [cassandra documentation](http://cassandra.apache.org/doc/latest/cql/ddl.html) section `CREATE KEYSPACE` for more information
+
+E.g.
+```
+CREATE KEYSPACE linagora_keyspace WITH REPLICATION = {'class' : 'SimpleStrategy', 'replication_factor' : 1};
+```
+
+#### Grant permissions on created keyspace to the role
+
+The role to be used by James need to have a full right on the keyspace 
+that James is using. Assuming the keyspace name is `linagora_keyspace` 
+and the role be `linagora`.
+
+```
+GRANT CREATE ON KEYSPACE linagora_keyspace TO linagora; // Permission to create tables on the appointed keyspace
+GRANT ALTER ON KEYSPACE linagora_keyspace TO linagora; // Permission to alter tables on the appointed keyspace and altering the keyspace itself
+GRANT ALTER ON ALL FUNCTIONS IN KEYSPACE linagora_keyspace TO linagora; // Permission to alter functions on the appointed keyspace
+GRANT DROP ON KEYSPACE linagora_keyspace TO linagora; // Permission to drop tables on the appointed keyspace
+GRANT DROP ON ALL FUNCTIONS IN KEYSPACE linagora_keyspace TO linagora; // Permission to drop functions on the appointed keyspace
+GRANT SELECT ON	KEYSPACE linagora_keyspace TO linagora; // Permission to select from tables on the appointed keyspace
+GRANT MODIFY ON	KEYSPACE linagora_keyspace TO linagora; // Permission to update data in tables on the appointed keyspace
+GRANT EXECUTE ON ALL FUNCTIONS IN KEYSPACE linagora_keyspace TO linagora; // Permission to execute functions on the appointed keyspace
+```
+
+**Tips**
+
+Since all of Cassandra roles using by different James are supposed to 
+have a same set of permissions. You can reduce the works by creating a 
+templating role like `typical_james_role` with all of necessary permissions. 
```

Comment:
is it a template (like: used at user creation) or a roleset (change existing user roles when you change it)?

Reply:
Oh yeah right, I don't know the answer and I couldn't find any resource mentioning this.

Reply:
Okay tested, it's roleset